lib/linter/apply-disable-directives.js

```diff
@@ -210,35 +209,61 @@ function applyDirectives(options) {
     const usedDisableDirectives = new Set();
 
     for (const problem of options.problems) {
+        let nextDirectiveIndex = 0;
```

Comment:
The code was optimized to go through the array of directives only once. Would it be possible to retain that?

If not, then do we need `disabledRuleMap` and `usedDisableDirectives`? Since the new version goes through all preceding directives for each problem, then it doesn't seem it needs to maintain this info for all rules.